src/ParameterLists.f90

```diff
@@ -2031,20 +2039,168 @@ SUBROUTINE getString_ParamType_scalar(thisParam,name,string,sskfmt,sdkfmt)
             !Error, can't do anything with a Plist.
           TYPE IS(ParamType_SSK)
             WRITE(tmpchar,TRIM(sskfmtDef)) param%val
+            string=TRIM(ADJUSTL(tmpchar))
           TYPE IS(ParamType_SDK)
             WRITE(tmpchar,TRIM(sdkfmtDef)) param%val
+            string=TRIM(ADJUSTL(tmpchar))
           TYPE IS(ParamType_SNK)
             WRITE(tmpchar,'(i0)') param%val
+            string=TRIM(ADJUSTL(tmpchar))
           TYPE IS(ParamType_SLK)
             WRITE(tmpchar,'(i0)') param%val
+            string=TRIM(ADJUSTL(tmpchar))
           TYPE IS(ParamType_SBK)
             WRITE(tmpchar,'(L1)') param%val
+            string=TRIM(ADJUSTL(tmpchar))
           TYPE IS(ParamType_STR)
-            tmpchar=param%val
+            string=param%val
+          TYPE IS(ParamType_SSK_a1)
+            WRITE(tmpchar,TRIM(sskfmtDef)) param%val(1)
+            string=TRIM(ADJUSTL(tmpchar))
+            DO i=2,SIZE(param%val)
+              WRITE(tmpchar,TRIM(sskfmtDef)) param%val(i)
+              string=string//delimDef//TRIM(ADJUSTL(tmpchar))
+            ENDDO
+          TYPE IS(ParamType_SDK_a1)
+            WRITE(tmpchar,TRIM(sdkfmtDef)) param%val(1)
+            string=TRIM(ADJUSTL(tmpchar))
+            DO i=2,SIZE(param%val)
+              WRITE(tmpchar,TRIM(sdkfmtDef)) param%val(i)
+              string=string//delimDef//TRIM(ADJUSTL(tmpchar))
+            ENDDO
+          TYPE IS(ParamType_SNK_a1)
+            WRITE(tmpchar,'(i0)') param%val(1)
+            string=TRIM(ADJUSTL(tmpchar))
+            DO i=2,SIZE(param%val)
+              WRITE(tmpchar,'(i0)') param%val(i)
+              string=string//delimDef//TRIM(ADJUSTL(tmpchar))
+            ENDDO
```

Comment:
It's worth noting that `IOUtils` has a `str(...)` function for integers and reals that eliminates the need for a temporary character string and, in many cases, formatting statements.  Might be good to start using that more for the SSK/SDK/SNK/SLK branches here.

Reply:
Ah, good to know Aaron.  I can go through an update those.  Do the SSK/SDK routines have an option for a format string?

Reply:
Yeah, there's a version that just returns all the digits, and a version that returns a specified number of decimal places.  I didn't look closely at how you're formatting the reals, so the `str` function may or may not work there.  But the integer version of `str` uses the `(i0)` formatting by default, so that one would definitely work.